### ISSUE TITLE
Parsing Error trying to run NodeJS tests

### ISSUE THREAD

**Issue by IPWright83** [wallabyjs/public#706]
### Issue description or question

I'm trying to test some code written in Node using ES6, but I'm getting a parser error when trying to run wallaby. The code however runs fine in Node and I thought I read that this should just run in wallaby? Do I need to pre-compile the output first?

> [Info]: Mon, 25 Jul 2016 08:28:01 GMT wallaby:project File routes/boards.js changes saved, store ts: 1469201644368, change ts: 1469435280608​​​​​
> ​​[Error]: Runtime error: SyntaxError: Parse error​​
> ​​​​​[Info]: Mon, 25 Jul 2016 08:28:01 GMT wallaby:project File tests/specs/boards.spec.js changes saved, store ts: 1469204011289, change ts: 1469435280619​​​​​
> ​​[Error]: at http://localhost:58896/mongo/mongo-rest.js?1469435280599&wallabyFileId=1:3​​
> ​​​​​[Info]: Mon, 25 Jul 2016 08:28:01 GMT wallaby:project Stopping process pool​​​​​
> ​​[Error]: Runtime error: SyntaxError: Parse error​​
> ​​​​​[Info]: Mon, 25 Jul 2016 08:28:01 GMT wallaby:project Running postprocessor​​​​​
> ​​​​​[Info]: Mon, 25 Jul 2016 08:28:01 GMT wallaby:postprocessor New TypeScript language service is required​​​​​
> ​​[Error]: at http://localhost:58896/routes/boards.js?1469435280608&wallabyFileId=4:3​​
> ​​​​​[Info]: Mon, 25 Jul 2016 08:28:01 GMT wallaby:project Postprocessor execution finished​​​​​
> ​​[Error]: Runtime error: ReferenceError: Can't find variable: require​​
> ​​[Error]: Runtime error: ReferenceError: Can't find variable: require​​
> ​​​​​[Info]: Mon, 25 Jul 2016 08:28:01 GMT wallaby:project Running all tests​​​​​

Here's an example of the code from the first file it mentions:

```
"use strict";

let colors = require('colors');
let ObjectID = require('mongodb').ObjectID;
let MongoWrapper = require('./mongo-wrapper');

/**
 * Used to handle all communication with Mongo in a RESTful way
 */
class MongoREST {

	/**
	 * Constructor for the MongoREST instance
	 * @param  {string} collection - The name of the collection that this REST client should be accessing
	 */
	constructor(collection) {
		this.collection = collection;
	}
}

module.exports = MongoREST;
```

### Wallaby.js configuration file

```
module.exports = function (wallaby) {
  return {
    files: [
        'mongo/*.js',
        'routes/*.js',
        'views/*.*'
    ],
    tests: [
    	"tests/specs/*.js"
    ],
    evn: {
      type: 'node',
      runner: 'node'
    },
    debug: true
  }
};
```

### Code editor or IDE name and version

Sublime Text v3 Build 3118

### OS name and version

Windows 10

**Comment by ArtemGovorov** [comment]
Looks like you have a typo in the `env` setting, so your tests are running in browser environment:
```javascript
    evn: {  // <--- typo
      type: 'node',
      runner: 'node'
    },
```

**Comment by IPWright83** [comment]
OK - that makes me sad. I've been playing around quite a lot (and even copied and pasted that config at one point), then obviously messed up putting it back in. Hopefully that'll get me sorted. Now gotta try and figure out where my gutter icons have gone again! Thanks @ArtemGovorov .

I really love WallabyJS when I'm using it, but I do feel like it's always a right pain to get setup (possibly because I only have superficial knowledge of things like Babel, Webpack etc so I'm never really 100% certain what I'm doing). I don't know if there's anything you can think of to help ease the process at all.

**Comment by IPWright83** [comment]
@ArtemGovorov Just going to run with this issue a little longer if that's ok... I'm up and running but got 2 things that are odd. 

Firstly I'm getting an error still in the console:

> [Error]: Runtime error: TypeError: Cannot set property '0' of undefined​​
> ​​[Error]:     at Object.<anonymous> (.\tests\specs\boards.spec.js:3:1)​​
> ​​[Error]:     at Module._compile (module.js:413:34)​​
> ​​[Error]:     at Object.Module._extensions..js (module.js:422:10)​​
> ​​[Error]:     at Module.load (module.js:357:32)​​
> ​​[Error]:     at Function.Module._load (module.js:314:12)​​
> ​​[Error]:     at Module.require (module.js:367:17)​​
> ​​[Error]:     at require (internal/module.js:20:19)​​

this is the source

```
"use strict";

describe("Boards Endpoint", function() {

	it("POST board 1", function() {
		expect(true).toBe(true);
	});

});
```

Secondly I get gutter icons if I open Sublime at the level where wallaby.js is at the root, but if it's down in a subdirectory it doesn't work. E.g. 

*Works*
![image](https://cloud.githubusercontent.com/assets/1374775/17095921/c71a0ec8-524e-11e6-98f3-0c3c4d4e4f4e.

*Doesn't work*

![image](https://cloud.githubusercontent.com/assets/1374775/17095932/d9d49c7c-524e-11e6-9714-aefc5a357843.png)

**Comment by ArtemGovorov** [comment]
> Runtime error: TypeError: Cannot set property '0' of undefined​​

Your test code looks ok, maybe just some caching issue. Does editing the config and restarting wallaby help? If the error doesn't go away and it reproduced consistently, please create a small repo where I could have a look.

> if it's down in a subdirectory it doesn't work

Did you select the wallaby config? When the config is in the opened root, wallaby can select the config  automatically by convention, when not, [you need to select it](https://wallabyjs.com/docs/intro/get-started-sublime.html#selecting-wallabyjs-config-file).

**Comment by IPWright83** [comment]
@ArtemGovorov indeed the first error was caching. Editing the config and restarting seems to flush stuff out (whereas I don't think just restarting did).

I have selected the wallaby.js config yes, and it's telling me that a single test ran.

**Comment by ArtemGovorov** [comment]
> I have selected the wallaby.js config yes, and it's telling me that a single test ran.

Is it not the expected result? Does it ignore some other tests that are running when wallaby config is in the opened root?

**Comment by IPWright83** [comment]
I've currently only got 2 sibling files. I've a single test in each which seems to be running fine in the test view and I can see the output. I could probably briefly screenshare if you wanted?

**Comment by ArtemGovorov** [comment]
> I've currently only got 2 sibling files. I've a single test in each which seems to be running fine in the test view and I can see the output. 

Sorry, I'm a bit lost. So everything works as expected? Is there still any issue with the config or we've figured everything out?

**Comment by IPWright83** [comment]
@ArtemGovorov sorry - I meant all the tests are running as you'd expect, no errors in the console, but no gutter-icons when `wallaby.js` is not at the root of my directory structure.

**Comment by ArtemGovorov** [comment]
Got it, thanks. Able to reproduce it now, looks like a bug, looking into it now, will post the update here.

**Comment by IPWright83** [comment]
Cool thanks @ArtemGovorov :) Glad I've been helpful.

**Comment by IPWright83** [comment]
@ArtemGovorov want me to create a new Github issue for you instead?

**Comment by ArtemGovorov** [comment]
@IPWright83 It's ok, I'll just change the original title.

**Comment by ArtemGovorov** [comment]
The issue is fixed in wallaby for Sublime Text package v1.0.22. The new version is published and should be available for updating to within an hour or so. @IPWright83 Thanks for your help!

**Comment by IPWright83** [comment]
@ArtemGovorov you're welcome. I'd use Wallaby for everything if I could, so great to help out with some bugs.
